packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +265,371 @@ export default class VrackMoveDialogCtrl {
+
+        // Update dedicated cloud data
+        this.updateDedicatedCloudServiceInfo();
+      }
+    }
+    if (result.dedicatedServer) {
+      if (
+        !this.data.eligibleServices.dedicatedServer ||
+        this.data.eligibleServices.dedicatedServer.length !==
+          result.dedicatedServer.length
+      ) {
+        this.data.eligibleServices.dedicatedServer = result.dedicatedServer;
+
+        // Update dedicated server data
+        this.updateDedicatedServerServiceInfo();
+      }
+    }
+    if (result.dedicatedServerInterface) {
+      if (
+        !this.data.eligibleServices.dedicatedServerInterface ||
+        this.data.eligibleServices.dedicatedServerInterface.length !==
+          result.dedicatedServerInterface.length
+      ) {
+        this.data.eligibleServices.dedicatedServerInterface =
+          result.dedicatedServerInterface;
+
+        // Update dedicated server interface
+        this.updateDedicatedServerInterfaceServiceInfo();
+      }
+    }
+    if (result.ip) {
+      if (
+        !this.data.eligibleServices.ip ||
+        this.data.eligibleServices.ip.length !== result.ip.length
+      ) {
+        this.data.eligibleServices.ip = result.ip;
+
+        // Update IP service
+        this.updateIpServiceInfo();
+      }
+    }
+    if (result.ipLoadbalancing) {
+      if (
+        !this.data.eligibleServices.ipLoadbalancing ||
+        this.data.eligibleServices.ipLoadbalancing.length !==
+          result.ipLoadbalancing.length
+      ) {
+        this.data.eligibleServices.ipLoadbalancing = result.ipLoadbalancing;
+
+        // Update IP Loadbalancing data
+        this.updateIpLoadbalancingServiceInfo();
+      }
+    }
+    if (result.ipv6) {
+      if (
+        !this.data.eligibleServices.ipv6 ||
+        this.data.eligibleServices.ipv6.length !== result.ipv6.length
+      ) {
+        this.data.eligibleServices.ipv6 = result.ipv6;
+
+        // Update IPv6
+        this.updateIpv6ServiceInfo();
+      }
+    }
+    if (result.legacyVrack) {
+      if (
+        !this.data.eligibleServices.legacyVrack ||
+        this.data.eligibleServices.legacyVrack.length !==
+          result.legacyVrack.length
+      ) {
+        this.data.eligibleServices.legacyVrack = result.legacyVrack;
+
+        // Update legacy vRack
+        this.updateLegacyVrackServiceInfo();
+      }
+    }
+  }
+
+  updateIpAndIpv6() {
+    if (this.data.eligibleServices?.ip?.length > 0) {
+      const ipv6List = this.data.eligibleServices.ip.filter((ip) =>
+        this.constructor.isIPv6(ip),
+      );
+      angular.forEach(ipv6List, (ipv6) => {
+        this.loadIpRegion(ipv6.id);
+      });
+    }
+  }
+
+  getEligibleServices() {
+    return this.vrackService
+      .getEligibleServices(this.serviceName)
+      .then(({ data }) => {
+        if (!this.data.eligibleServices) {
+          this.data.eligibleServices = data.result;
+
+          // Update eligible services infos for each service not empty
+          if (
```

Comment:
Yes it could be, but in the same time we could have many service should be update and if I use a switch case, I could update only one service. it's not a "if... else if..." but a "if... if"

Reply:
getEligibleServices and updateDedicatedCloudServiceInfo are not easy to read since too big.

Maybe a trick would be first to separate the logic of the updateDedicatedCloudServiceInfo  into several dedicated updateDedicatedCloudServiceInfo{Suffix} functions where you implement a logic for the given suffix.

Then a refactor on the common keys to avoid the `if if if if if` since you will have dedicated function, so it will fit the switch/case proposed above